### DISCUSSION TITLE
Final DB users

### DISCUSSION THREAD

**Discussion by marcpley** [pawelsalawa/sqlitestudio#5157] [Ideas]
Pawel said in another discussion that SQLiteStudio "is a tool aimed at database professionals and enthusiasts".
Is SQLiteStudio intended solely for DB developers or also for final users of these DBs when it's possible?
When a DB don't really need a spécific app, using the DB directly in SQLiteStudio save the big big dev time off the app.
Final DB usage need specific editing and configuring features.

I'd like to know whether this orientation is irrelevant or not :-)

**Comment by pawelsalawa** [comment]
Well, it depends who do you mean by "final users". I imagine that such people usually also need to have decent skills with databases, don't they?

**Comment by marcpley** [comment]
No in my opinion. They know they have to manage data in tables and found answers in these tables. Some columns are automatically filed, other they must input the data in. They know how to sort or filter the data but don't know what is a trigger or a foreign key.
For such users I would like to (more to less important) : 
1- help then to quickly input/update the data (that is already very good with 3.4.8)
2- help then to quickly read and understand the data (#5078 and other ideas)
3- lock/hide the structure modification
4- add buttons to fire SQL files or enhance the actual SQL window, plus a richer param input dialog
5- provide inline help for the database
6- distribute a set with database, functions, SQL files and help 

This pseudo software can be distributed to few users because it can be developed far more rapidly and easilly  than a cross-platform software.

I can develop that in specifics issues or discussions... if you are ok to move forward in this area.

**Reply by pawelsalawa** [reply]
I see. In that case I say SQLiteStudio is not for such people. End-user features are subject for end-user applications. Of course SQLiteStudio could address some common features, but this would require significantly more application adjustments and options, which always means more effort and time (which I obviously don't have enough). I'd rather focus on features intended for DB developers.

**Comment by marcpley** [comment]
What about developing that via plugins ?
If other than you spend time for that.

**Reply by pawelsalawa** [reply]
I'm not sure how feasible it is to do it through plugins. It seems to require significant changes to the core application.

Anyway, plugins are always up to whoever wants to develop them. As I pointed out, I'm focusing on the primary audience, which are people developing databases and I dedicate my time to this direction.

**Comment by marcpley** [comment]
I can dedicate time to the plugin feasible enhancements.

**Comment by marcpley** [comment]
but I understand that will need time to you to guide me.

**Reply by pawelsalawa** [reply]
Sure, I will guide you.

**Comment by marcpley** [comment]
Cool. I will create issues and could mark them "plugins feasible" or not.
I've changed my computer, I have to reinstall Qt, what is the appropriate version now?

**Reply by pawelsalawa** [reply]
For SQLiteStudio 3.5.x the Qt is going to be at least 6.2. I'm currently using 6.8.1 on my PC.
